### PR TITLE
Avoid using raw pointers in GetAllNewerThan results

### PR DESCRIPTION
**Pull request by UdjinM6**
I believe this fixed some crashes for me in GUI (with Governance tab on).

### CODE REVIEW

**Comment by UdjinM6** [comment]
oh, wow! that's a lot of conflicting PRs for what I thought was a simple change 🙈

**Comment by PastaPastaPasta** [comment]
Can you convert this to using weak ptr? These really shouldn't be owning

**Comment by UdjinM6** [comment]
> Can you convert this to using weak ptr? These really shouldn't be owning

I'm not sure how to do this without a lot of refactoring in CGovernanceManager. A suggestion (or a post-merge followup) patch is welcome :)

**Comment by PastaPastaPasta** [comment]
how about smth like https://github.com/dashpay/dash/commit/632179859c8a95790c8695529782a2912663645a to avoid ownership? (shared_ptr changes may still be appropriate to prevent dangling, although I don't like it)

**Comment by PastaPastaPasta** [comment]
Also is https://github.com/dashpay/dash/blob/632179859c8a95790c8695529782a2912663645a/src/qt/governancelist.cpp#L342 a mem leak? @UdjinM6 @5tefan

(please explain if I'm missing smth)

**Comment by PastaPastaPasta** [comment]
I would propose a solution like 8a092644ec1d1e71b92349f0b6848dbab8f31778 as opposed to introducing shared_ptr usage here. It basically extends the lifetime of the lock, to the lifetime of the returned vector (whose values should be considered dangling after the lock expires). as opposed to creating some 40 shared_ptrs (relatively expensive) and extending the lifetime of the governance objects themselves, we only extend the time that we were locking the mutex. (although this ReturnWithLock struct maybe should have a different name, and maybe be in a different file)

File: src/governance/governance.cpp [hunk 1]
```diff
@@ -516,8 +516,7 @@ std::vector<const CGovernanceObject*> CGovernanceManager::GetAllNewerThan(int64_
         }
 
         // ADD GOVERNANCE OBJECT TO LIST
-        const CGovernanceObject* pGovObj = &(objPair.second);
-        vGovObjs.push_back(pGovObj);
+        vGovObjs.push_back(std::make_shared<const CGovernanceObject>(objPair.second));
```
Comment (PastaPastaPasta) [inline src/governance/governance.cpp:519]: This is a bad idea anyway. If I'm understanding correctly, right here you're creating a copy, creating a shared_ptr to that copy, and returning the vector of those shared_ptrs. You could simply return a vector of copies if you don't like my lock extension idea

**Comment by PastaPastaPasta** [comment]
Any update on this?

**Comment by PastaPastaPasta** [comment]
re, any update on this?

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**